aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -107,55 +108,49 @@ class BasePersistenceLayer(ABC):
     Abstract Base Class for Idempotency persistence layer.
     """
 
-    def __init__(
-        self,
-        event_key_jmespath: str = "",
-        payload_validation_jmespath: str = "",
-        expires_after_seconds: int = 60 * 60,  # 1 hour default
-        use_local_cache: bool = False,
-        local_cache_max_items: int = 256,
-        hash_function: str = "md5",
-        raise_on_no_idempotency_key: bool = False,
-        jmespath_options: Dict = None,
-    ) -> None:
+    def __init__(self):
+        """Initialize the defaults """
+        self.configured = False
+        self.event_key_jmespath: Optional[str] = None
+        self.event_key_compiled_jmespath = None
+        self.jmespath_options: Optional[dict] = None
+        self.payload_validation_enabled = False
+        self.validation_key_jmespath = None
+        self.raise_on_no_idempotency_key = False
+        self.expires_after_seconds: int = 60 * 60  # 1 hour default
+        self.use_local_cache = False
+        self._cache: Optional[LRUDict] = None
+        self.hash_function = None
+
+    def configure(self, config: IdempotencyConfig) -> None:
```

Comment:
I was thinking of having this in the constructor and change the way we call it within IdempotencyHandler, but honestly that can be done later - there are a few areas we'll need to refactor, and this will be an internal change either way, and in the worst case we're calling it Beta too, it'll be a minor change IF that.